Doc/library/asyncio-dev.rst

```diff
@@ -248,3 +248,104 @@ Output in debug mode::
       File "../t.py", line 4, in bug
         raise Exception("not consumed")
     Exception: not consumed
+
+
+Asynchronous generators best practices
+======================================
+
+By :term:`asynchronous generator` in this section we will mean
+an :term:`asynchronous generator iterator` that returned by
+:term:`asynchronous generator` function.
+
+Manually close the generator
+----------------------------
+
+If an asynchronous generator happens to exit early by :keyword:`break`, the caller
+task being cancelled, or other exceptions, the generator's async cleanup code
+will run and possibly raise exceptions or access context variables in an
+unexpected context--perhaps after the lifetime of tasks it depends, or
+during the event loop shutdown when the async-generator garbage collection hook
+is called.
+
+To prevent this, it is recommended to explicitly close the async generator by
+calling :meth:`~agen.aclose` method, or using :func:`contextlib.aclosing` context
+manager::
+
+  import asyncio
+  import contextlib
+
+  async def gen():
+    yield 1
+    yield 2
+
+  async def func():
+    async with contextlib.aclosing(gen()) as g:
+      async for x in g:
+        break
+
+  asyncio.run(func())
+
+Only create a generator when a loop is already running
+------------------------------------------------------
+
+As said above, if an asynchronous generator is not resumed before it is
+finalized, then any finalization procedures will be delayed. The event loop
+handles this situation and doing it best to call async generator-iterator's
+:meth:`~agen.aclose` at the proper moment, thus allowing any pending
+:keyword:`!finally` clauses to execute.
+
+Then it is recomended to create async generators only after the event loop
+has already been created.
```

Comment:
I'm not sure I follow how it follows from the previous paragraph (which describes that finalization of an async generator-iterator may be delayed if its body hasn't finished yet) to the recommendation of creating them only after an event loop exists.